components/cdc/src/delegate.rs

```diff
@@ -61,6 +66,40 @@ impl Default for DownstreamState {
     }
 }
 
+/// Shold only be called when it's uninitialized or stopped. Return false if it's stopped.
+pub fn on_init_downstream(s: &AtomicCell<DownstreamState>) -> bool {
```

Comment:
I think the visibility of these methods could be smaller, for example they should be pub(crate). Using smaller visibility would help us to find dead code.